posts/2019-08-12-10-tips-creating-robust-serverless-components.md

```diff
@@ -0,0 +1,88 @@
+---
+title: "10 Tips for Creating Robust Serverless Components"
+description: "A practical guide for creating robust, safe and fast serverless components."
+date: 2019-08-07
+thumbnail: "https://s3-us-west-2.amazonaws.com/assets.blog.serverless.com/serverless-components-beta/serverless-components-thumbnail.gif"
+heroImage: "https://s3-us-west-2.amazonaws.com/assets.blog.serverless.com/serverless-components-beta/serverless_components_header.gif"
+category:
+  - guides-and-tutorials
+authors:
+  - EslamHefnawy
+---
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/ts26BVuX3j0" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+[Serverless Components](https://github.com/serverless/components) allow anyone to design and build specific use cases and share them with the world, without having to be tied to a single pattern or a single opinion. We then released 20+ components to bootstrap these uses cases, and we’re constantly building more. One of our core design goals is simplicity, so we produced an incredibly simple core and components API that makes creating components a breeze, and we’ve covered them all in a [simple one-page documentation](https://github.com/serverless/components/blob/master/README.md).
+
+In this article, we will give you comprehensive practice advice (not docs) on creating robust Serverless Components based on our experience building all these components over the past year, to help as you start creating components for your own unique use cases. So let’s dig in..
+
+### 1. Start with the Outcome
+
+Serverless Components is an outcomes framework. Every single component produces a specific outcome. Those outcomes could be low-level (like an S3 bucket), or high level (like a complete cloud storage solution). Before you even start building your component, you need to decide on what that outcome is and how much abstraction you want to provide for the best UX.
```

Comment:
+1 "Serverless Components is an outcomes framework"

Suggestion:  Change (like a complete cloud storage solution) to (an image processing feature).